src/simulation/m_time_steppers.fpp

```diff
@@ -892,6 +910,127 @@ contains
+        real(kind(0d0)) :: vel_sum    !< Cell-avg. velocity sum
+        real(kind(0d0)) :: pres       !< Cell-avg. pressure
+        real(kind(0d0)), dimension(num_fluids) :: alpha      !< Cell-avg. volume fraction
+        real(kind(0d0)) :: gamma      !< Cell-avg. sp. heat ratio
+        real(kind(0d0)) :: pi_inf     !< Cell-avg. liquid stiffness function
+        real(kind(0d0)) :: qv         !< Cell-avg. fluid reference energy
+        real(kind(0d0)) :: c          !< Cell-avg. sound speed
+        real(kind(0d0)) :: E          !< Cell-avg. energy
+        real(kind(0d0)) :: H          !< Cell-avg. enthalpy
+        real(kind(0d0)), dimension(2) :: Re         !< Cell-avg. Reynolds numbers
+        real(kind(0d0)) :: blkmod1, blkmod2 !<
+            !! Fluid bulk modulus for Woods mixture sound speed
+        type(int_bounds_info) :: ix, iy, iz
+        type(vector_field) :: gm_alpha_qp
+
+        integer :: i, j, k, l, q !< Generic loop iterators
+
+        real(kind(0d0)) :: dt_local
+
+        integer :: Nfq
+        real(kind(0d0)) :: fltr_dtheta   !<
+         !! Modified dtheta accounting for Fourier filtering in azimuthal direction.
+
+        ix%beg = 0; iy%beg = 0; iz%beg = 0
+        ix%end = m; iy%end = n; iz%end = p
+        call s_convert_conservative_to_primitive_variables( &
+            q_cons_ts(1)%vf, &
+            q_prim_vf, &
+            gm_alpha_qp%vf, &
+            ix, iy, iz)
+
+        ! Computing Stability Criteria at Current Time-step ================
+        !$acc parallel loop collapse(3) gang vector default(present) private(alpha_rho, vel, alpha, Re, fltr_dtheta, Nfq)
+        do l = 0, p
+            do k = 0, n
+                do j = 0, m
+
+                    do i = 1, num_fluids
+                        alpha_rho(i) = q_prim_vf(i)%sf(j, k, l)
+                        alpha(i) = q_prim_vf(E_idx + i)%sf(j, k, l)
+                    end do
+
+                    if (bubbles) then
+                        call s_convert_species_to_mixture_variables_bubbles_acc(rho, gamma, pi_inf, qv, alpha, alpha_rho, Re, j, k, l)
+                    else
+                        call s_convert_species_to_mixture_variables_acc(rho, gamma, pi_inf, qv, alpha, alpha_rho, Re, j, k, l)
+                    end if
+
+                    do i = 1, num_dims
+                        vel(i) = q_prim_vf(contxe + i)%sf(j, k, l)
+                    end do
+
+                    vel_sum = 0d0
+                    do i = 1, num_dims
+                        vel_sum = vel_sum + vel(i)**2d0
+                    end do
+
+                    pres = q_prim_vf(E_idx)%sf(j, k, l)
+
+                    E = gamma*pres + pi_inf + 5d-1*rho*vel_sum + qv
+
+                    H = (E + pres)/rho
+
+                    ! Compute mixture sound speed
+                    call s_compute_speed_of_sound(pres, rho, gamma, pi_inf, H, alpha, vel_sum, c)
+
+                    if (grid_geometry == 3) then
+                        if (k == 0) then
```

Comment:
i suspect this logic already exists somewhere else in the code?

Reply:
This logic is copied from `m_data_outpu` where the CFL is calculated given a specific delta t, but computing a delta t given a CFL number requires separate logic. At east as far as I can tell.

Reply:
i see. maybe we need two subroutines that can be used everywhere: (1) compute CFL from `dt` and (2) compute `dt` from CFL.

Reply:
Ok